sources/logdog/src/commands.rs

```diff
@@ -0,0 +1,154 @@
+//! Provides the list of commands that `logdog` will run and provides the handler function to run
+//! them.
+//!
+//! # Static Commands
+//!
+//! At build time, the `build.rs` file checks which variant is being built and creates a symlink
+//! file which points to the commands for the current variant. This file is named `variant`. We load
+//! the `variant` and `common` files into static strings at compile time, and these provide the list
+//! of commands that `logdog` will run.
+
+use crate::error::{self, Result};
+use reqwest::blocking::{Client, Response};
+use snafu::{OptionExt, ResultExt};
+use std::fs::File;
+use std::path::Path;
+use std::process::Stdio;
+use url::Url;
+
+/// The `logdog` commands that all variants have in common.
+const COMMON_COMMANDS: &str = std::include_str!("../static/common");
+/// The `logdog` commands that are specific to the current variant.
+const VARIANT_COMMANDS: &str = std::include_str!("../static/variant");
+
+/// Returns the list of commands to run by combining `COMMON_COMMANDS` and `VARIANT_COMMANDS`.
+/// These are read at compile-time from files named `common` and `variant` respectively.
+pub(crate) fn commands() -> Vec<&'static str> {
+    let mut commands: Vec<&str> = COMMON_COMMANDS
+        .lines()
+        .filter(|&command| !command.is_empty())
+        .collect();
+    let mut variant_commands: Vec<&str> = VARIANT_COMMANDS
+        .lines()
+        .filter(|&command| !command.is_empty())
+        .collect();
+    commands.append(&mut variant_commands);
+    commands
+}
+
+/// A logdog `Command` is in the format `mode filename [args...]`. `mode` specifies what type of
+/// command it is, e.g. `sh` for a shell command or `http` for an HTTP get request. `filename` is
+/// the name of the output file. `args` is any number of shell-like arguments. Putting this all
```

Comment:
```suggestion
/// the name of the output file. `args` is any number of shell-like arguments, for `sh`, or a URL, for `http`. Putting this all
```
I guess "shell-like" doesn't apply to the http arguments...

Reply:
[afdc876](https://github.com/bottlerocket-os/bottlerocket/compare/6eb611d17506ab4aad98e5db4cd17b00bacd0839..afdc8765a9e97bb1d4436eda9b0c964fce24ab95)